src/vs/workbench/contrib/terminal/node/terminalProcess.ts

```diff
@@ -69,28 +69,32 @@ export class TerminalProcess implements ITerminalChildProcess, IDisposable {
 			experimentalUseConpty: useConpty
 		};
 
-		try {
-			this._ptyProcess = pty.spawn(shellLaunchConfig.executable!, shellLaunchConfig.args || [], options);
-			this._processStartupComplete = new Promise<void>(c => {
-				this.onProcessIdReady((pid) => {
-					c();
+		const filePath = path.basename(shellLaunchConfig.executable!);
+		fs.stat(filePath, (err, stats) => {
+			if (err === null) {
+				this._ptyProcess = pty.spawn(shellLaunchConfig.executable!, shellLaunchConfig.args || [], options);
+				this._processStartupComplete = new Promise<void>(c => {
+					this.onProcessIdReady((pid) => {
+						c();
+					});
 				});
-			});
-		} catch (error) {
-			// The only time this is expected to happen is when the file specified to launch with does not exist.
-			this._exitCode = 2;
-			this._queueProcessExit();
-			this._processStartupComplete = Promise.resolve(undefined);
-			return;
-		}
-		this._ptyProcess.on('data', (data) => {
+			}
+			else if (err.code === 'ENOENT') {
```

Comment:
Let's handle the error cases first to reduce indentation:

```ts
if (err && err.code === 'ENOENT') {
  ...
  return;
}

// spawn...
```

Reply:
Yes this makes sense